### PR TITLE
Added distributed locks documentation to js-sdk

### PR DESCRIPTION
**Pull request by amulyavarote**
Signed-off-by: Amulya Varote <amulyavarote@microsoft.com>

# Description

_Please explain the changes you've made_

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #350 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

**Review by hhunter-ms — DISMISSED** [review]
lgtm

File: daprdocs/content/en/js-sdk-docs/js-client/_index.md [hunk 1]
```diff
@@ -387,6 +387,70 @@ start().catch((e) => {
...
+  // Checking if the lock exists.
+  console.log(`Unlocking on ${storeName}, ${resourceId} as owner: ${lockOwner}`);
+  const lockDoesNotExistResponse = await client.lock.unlock(storeName, resourceId, lockOwner);
+  console.log("Unlock API response when lock is not acquired: " + getResponseStatus(lockDoesNotExistResponse.status));
+
+  expiryInSeconds = 25;
+  console.log(`Acquiring lock on ${storeName}, ${resourceId} as owner: ${lockOwner}`);
+  const tryLockResponse1 = await client.lock.tryLock(storeName, resourceId, lockOwner, expiryInSeconds);
+  console.log("Acquired Lock? " + tryLockResponse1.success);
+
```
Comment (shubham1172) [inline daprdocs/content/en/js-sdk-docs/js-client/_index.md:430]: Recommendation to remove this. The APIs are already introduced before. This is showing a behavior, which we can skip

File: daprdocs/content/en/js-sdk-docs/js-client/_index.md [hunk 2]
```diff
@@ -387,6 +387,70 @@ start().catch((e) => {
...
+const daprHost = "127.0.0.1";
+const daprPortDefault = "3500";
+
+async function start() {
+  const client = new DaprClient(
+    daprHost,
+    process.env.DAPR_GRPC_PORT ?? daprPortDefault,
+    CommunicationProtocolEnum.GRPC
+  );
```
Comment (shubham1172) [inline daprdocs/content/en/js-sdk-docs/js-client/_index.md:406]: We can simplify this.
```suggestion
  const client = new DaprClient(daprHost, daprPort);
```

**Review by shubham1172 — APPROVED** [review]

**Comment by shubham1172** [comment]
Thank you @amulyavarote for the quick PR! 🎉

**Pull request merged**